src/material/select/select.ts

```diff
@@ -888,7 +888,10 @@ export abstract class _MatSelectBase<C> extends _MatSelectMixinBase implements A
     merge(...this.options.map(option => option._stateChanges))
       .pipe(takeUntil(changedOrDestroyed))
       .subscribe(() => {
-        this._changeDetectorRef.markForCheck();
+        // `_stateChanges` can fire as a result of a change in the label's DOM value which may
+        // be the result of an expression changing. We have to use `detectChanges` in order
+        // to avoid "changed after checked" errors (see #14793).
+        this._changeDetectorRef.detectChanges();
```

Comment:
What causes `stateChanges` to emit after dirty checking occurs? Are we using a mutation observer or something similar?

Reply:
We're using an `ngAfterViewChecked` hook that looks at the `textContent` of the option and emits when it changes.

Reply:
What if we used `cdkObserveContent` instead?

Reply:
That would work too, but I'm hesitant to apply it on each option since people tend to render lots of them in a single select.

Reply:
We could debate changing how we do it, but I think we should at least make this change for now. It maintains the same approach but fixes the error.